engine/common/version/version_control_test.go

```diff
@@ -0,0 +1,312 @@
+package version
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/stretchr/testify/assert"
+	testifyMock "github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	storageMock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/onflow/flow-go/utils/unittest/mocks"
+)
+
+// TestInitialization tests the initialization process of the VersionControl component.
+func TestInitialization(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	finalizedRootBlockHeight := uint64(0)
+	sealedHeader := unittest.BlockHeaderFixture(unittest.WithHeaderHeight(10))
+
+	t.Run("happy case", func(t *testing.T) {
```

Comment:
is there a way to write these as table driven tests? e.g. a block of config at the top that defines the inputs/expected output, then a single path of test logic. This makes it much easier to reason about which tests are included and if the corner cases are handled.